src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -756,21 +819,26 @@ private static int compareImports(String import1, String import2) {
     }
 
     /**
-     * Counts empty lines before given.
-     * @param lineNo
-     *        Line number of current import.
-     * @return count of empty lines before given.
+     * Counts empty lines between given parameters.
+     * @param fromLineNo
+     *        One-based line number of previous import.
+     * @param toLineNo
+     *        One-based line number of current import.
+     * @return count of empty lines between given parameters.
      */
-    private int getCountOfEmptyLinesBefore(int lineNo) {
+    private int getCountOfEmptyLinesBetween(int fromLineNo, int toLineNo) {
         int result = 0;
         final String[] lines = getLines();
-        //  [lineNo - 2] is the number of the previous line
-        //  because the numbering starts from zero.
-        int lineBeforeIndex = lineNo - 2;
-        while (lineBeforeIndex >= 0
-                && CommonUtil.isBlank(lines[lineBeforeIndex])) {
-            lineBeforeIndex--;
-            result++;
+
+        //  "- 2" because the numbering starts from zero
+        for (int i = toLineNo - 2; i > fromLineNo - 2; i--) {
+            if (CommonUtil.isBlank(lines[i])) {
+                result++;
+            }
+            else {
+                // stop on none empty line to keep the behavior same as previous
```

Comment:
Should it be "stop on non-empty line..."?

Reply:
Good catch :)

Reply:
Fixed.

Reply:
@gmshake , please share link to test that cover this catch.

Reply:
just comment is changed